backends/dpdk/dpdkMetadata.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 Intel Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "dpdkMetadata.h"
+#include "dpdkUtils.h"
+
+namespace DPDK {
+const IR::Node* AddNewMetadataFields::preorder(IR::DpdkStructType *st) {
+    if (isMetadataStruct(st)) {
+        for (auto nf : newMetadataFields) {
+                st->fields.push_back(nf);
+        }
+        auto newSt = new IR::DpdkStructType(st->srcInfo, st->name,
+                                          st->annotations, st->fields);
+        newMetadataFields.clear();
+        return newSt;
+    }
+    return st;
+}
+
+// make sure new decls and fields name are unique
+void DirectionToRegRead::uniqueNames(IR::DpdkAsmProgram *p) {
+    P4::MinimalNameGenerator mng;
+    for (auto st : p->structType) {
+        if (isMetadataStruct(st)) {
+            for (auto field : st->fields)
+                mng.usedName(field->name);
+        }
+    }
+    reg_read_tmp = mng.newName("reg_read_tmp");
+    left_shift_tmp = mng.newName("left_shift_tmp");
+    P4::MinimalNameGenerator mng1;
+    for (auto decl : p->externDeclarations) {
+        mng1.usedName(decl->name);
+    }
+    registerInstanceName = mng1.newName("network_port_mask");
```

Comment:
network_port_mask is to be treated like a reserved name. For this we don't need to use name generator.
Also, it would be good to add a comment that this name is reserved by DPDK target and should not be used by the P4 programmer.
We can issue an error if it already exists in the P4 program.